rig/routing_table/ordered_covering.py

```diff
@@ -467,33 +472,45 @@ def apply(self, aliases):
         {(key, mask): {(key, mask), ...}}
             A new aliases dictionary.
         """
-        # Generate a new copy of the routing table
-        routing_entries = list(self.routing_table)
+        # Create a new routing table of the correct size
+        new_size = len(self.routing_table) - len(self.entries) + 1
+        new_table = [None for _ in range(new_size)]
+
+        # Create a copy of the aliases dictionary
         aliases = dict(aliases)
 
-        # Add the new entry
+        # Get the new entry
         new_entry = RoutingTableEntry(
-            route=routing_entries[next(iter(self.entries))].route,
+            route=self.routing_table[next(iter(self.entries))].route,
             key=self.key, mask=self.mask, sources=self.sources
         )
-        routing_entries.insert(self.insertion_index, new_entry)
         aliases[(self.key, self.mask)] = our_aliases = set([])
 
-        # Remove all the old entries.
-        # NOTE: This relies on the fact that merging can only ever increase
-        # generality: the entry that was just inserted MUST be below the
-        # entries that we need to remove and hence doesn't affect their
-        # indices.
-        offset = 0
-        for i in sorted(self.entries):
-            entry = routing_entries.pop(i - offset)
-            offset += 1
-            km = (entry.key, entry.mask)
-
-            # Add to the alias list
-            our_aliases.update(aliases.pop(km, set([km])))
-
-        return routing_entries, aliases
+        # Iterate through the old table copying entries acrosss
+        insert = 0
+        for remove, entry in enumerate(self.routing_table):
```

Comment:
I think I would prefer "remove" to be "i" since "remove" does not say "index in the (old) table" to me.